### PR TITLE
make encryption configurable for home storage

### LINKED ISSUE #19061: Encryption setting to disable encryption for the home storage
The use case of server-side encryption is to protected the users against malicious 3rd party storage providers. The user can already enable/disable encryption for different external storages but he can't disable encryption for the home storage. With the use case of server-side encryption in mind it would make a lot of sense to disable encryption for the home storage and only enable it for some external storages.

Therefore I suggest a setting to disable encryption for the home storage. This was also suggested by multiple users during the ownCloud conference. We still need to decide if it makes more sense to have such a setting in core or on a per-encryption-module level. Opinions? Suggestions?

cc @DeepDiver1975 @cmonteroluque @karlitschek @jknockaert

### PR DESCRIPTION
**Pull request by schiessle**
Let the admin decide if the home storage should be encrypted or not. This way encryption can be use explicitly for mounted external storages.

If you want to try it out, this are the steps:
1. enable server side encryption and the ownCloud default encryption module
2. relogin
3. go to admin settings
4. now you see a checkbox to enable/disable encryption of the home storage

-> if it is checked files in the home storage should be encrypted, if it is not checked encryption should skip all files stored in the home storage

fix https://github.com/owncloud/core/issues/19061

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@schiesbn to review?

**Comment by schiesbn** [comment]
@MorrisJobke sure :smile:

File: apps/encryption/templates/settings-admin.php [hunk 1]
```diff
@@ -9,56 +9,63 @@
...
-		<input
-			type="password"
-			name="changeRecoveryPassword"
-			id="oldEncryptionRecoveryPassword"
-			placeholder="<?php p($l->t("Old recovery key password")); ?>"/>
+		<p id="encryptHomeStorageSetting">
+			<input type="checkbox" name="encrypt_home_storage" id="encryptHomeStorage"
+				   value="1" <?php if ($_['encryptHomeStorage']) print_unescaped('checked="checked"'); ?> />
+			<label for="encryptHomeStorage"><?php p($l->t('Encrypt the home storage'));?></label></br>
+			<em><?php p( $l->t( "Enabling this option encrypt all files stored on the main storage, otherwise only files on external storage will be encrypted" ) ); ?></em>
```
Comment (PVince81) [inline apps/encryption/templates/settings-admin.php]: => "encrypt**s**"

Comment (PVince81) [inline apps/encryption/templates/settings-admin.php]: or "will encrypt"

**Comment by PVince81** [comment]
Tested, works :+1: 
Shared storage and external storage still work independently from this setting.

**Comment by schiesbn** [comment]
second reviewer... maybe @MorrisJobke @nickvergessen or @LukasReschke ? Thanks!

File: apps/encryption/controller/settingscontroller.php [hunk 2]
```diff
@@ -143,4 +151,14 @@ public function updatePrivateKeyPassword($oldPassword, $newPassword) {
 		}
 
 	}
+
+	/**
+	 * @UseSession
+	 *
+	 * @param bool $encryptHomeStorage
+	 * @return DataResponse
```
Comment (nickvergessen) [inline apps/encryption/controller/settingscontroller.php]: I can't see that

Comment (schiesbn) [inline apps/encryption/controller/settingscontroller.php]: > I can't see that

added

File: apps/encryption/templates/settings-admin.php [hunk 3]
```diff
@@ -9,56 +9,63 @@
...
-		<?php p($l->t("Change recovery key password:")); ?>
-		<span class="msg"></span>
-		<br/>
-		<input
-			type="password"
-			name="changeRecoveryPassword"
-			id="oldEncryptionRecoveryPassword"
-			placeholder="<?php p($l->t("Old recovery key password")); ?>"/>
+		<p id="encryptHomeStorageSetting">
+			<input type="checkbox" name="encrypt_home_storage" id="encryptHomeStorage"
```
Comment (nickvergessen) [inline apps/encryption/templates/settings-admin.php]: add `class="checkbox"` for magic beauty

**Comment by nickvergessen** [comment]
:+1: works as described

File: apps/encryption/templates/settings-admin.php (hunk 3 above)
Comment (schiesbn) [inline apps/encryption/templates/settings-admin.php]: done

**Comment by b1ggjoe** [comment]
So will this be back-ported to current stable 8.2 downloads or will this be built-in in a future update?

Thanks,

BJ

**Comment by PVince81** [comment]
New feature, so 9.0 only.

Only (critical) bugfixes get backported.

File: apps/encryption/lib/util.php [hunk 4]
```diff
@@ -94,12 +94,41 @@ public function isRecoveryEnabledForUser($uid) {
...
+			'encryption',
+			'encryptHomeStorage',
+			'1'
+		);
+
+		return ($encryptHomeStorage === '1');
+	}
+
+	/**
+	 * check if the home storage should be encrypted
```
Comment (phil-davis) [inline apps/encryption/lib/util.php]: Description here is for "check" but this is the "set" function.

Comment (schiesbn) [inline apps/encryption/lib/util.php]: good catch, do you want to create a pull request to fix it?

Comment (phil-davis) [inline apps/encryption/lib/util.php]: PR https://github.com/owncloud/core/pull/20272

**Comment by carlaschroder** [comment]
This is good stuff. Nice work.

**Pull request merged**